homeassistant/components/media_player/vieratv.py

```diff
@@ -0,0 +1,249 @@
+"""
+Support for interface with a Panasonic Viera TV.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.vieratv/
+"""
+import logging
+import socket
+import xml.etree.ElementTree as ET
+from urllib.request import urlopen, Request
+
+from homeassistant.components.media_player import (
+    DOMAIN, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_TURN_OFF, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP, MediaPlayerDevice)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, STATE_OFF, STATE_ON, STATE_UNKNOWN)
+from homeassistant.helpers import validate_config
+
+CONF_PORT = "port"
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = []
+
+SUPPORT_VIERATV = SUPPORT_PAUSE | SUPPORT_VOLUME_STEP | \
+    SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+    SUPPORT_TURN_OFF
+
+URN_RENDERING_CONTROL = 'schemas-upnp-org:service:RenderingControl:1'
+URN_REMOTE_CONTROL = 'panasonic-com:service:p00NetworkControl:1'
+
+URL_CONTROL_DMR = 'dmr/control_0'
+URL_CONTROL_NRC = 'nrc/control_0'
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Panasonic Viera TV platform."""
+    # Validate that all required config options are given
+    if not validate_config({DOMAIN: config}, {DOMAIN: [CONF_HOST]}, _LOGGER):
+        return False
+
+    host = config.get(CONF_HOST, None)
+    port = config.get(CONF_PORT, 55000)
+    name = config.get(CONF_NAME, 'Panasonic Viera TV')
+
+    remote = PanasonicVieraTVRemoteControl(host, port)
+
+    add_devices([PanasonicVieraTVDevice(name, remote)])
+    return True
+
+
+class PanasonicVieraTVRemoteControl:
```

Comment:
Can you please extract the API code to interact with the TV into a separate package that Home Assistant can depend on?